ZENKO-3737 - MongoDB dashboard

jbertran opened the pull request:
* Add a MongoDB dashboard
* Add a README for future contributors to dashboards
* Enforce JSON dashboard version increments in CI

NOTE: this PR introduces the expectations that the version of a dashboard is the number of commits that change it. If you get a failure in CI, this might be the culprit.

Commit:
ZENKO-3737 - provide README for dashboard management

Commit:
ZENKO-3737 - add MongoDB Grafana dashboard

From https://grafana.com/grafana/dashboards/12079

ZENKO-3737 - switch to `namespace` input

ZENKO-3737 - remove annotations from mongodb dashboard

Commit:
ZENKO-3737 - enforce dashboard version increments in CI

bert-e commented:
# Hello jbertran,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

lavigne958 commented on monitoring/dashboards/README.md:
```diff
@@ -0,0 +1,30 @@
+2. Apply the following modifications:
+   * add the `__inputs` of the original download to the model
+   * add the `__requires` of the original download to the model
+   * move `id` from int to `null` in the model
```
I would add: `* remove "uid" entry`

jbertran replied:
Is this OK? I remember we weren't sure about removing or leaving as is yesterday on Slack.

francoisferrand replied:
I set it to `null`, like `id`...

lavigne958 replied:
I don't see any mentions of *UID* but only *ID* did you mixed them up perhaps ?

francoisferrand replied:
nop, there are 2 fields. But maybe it depends on where the dashboard comes from, or the version...
(See lines 109 and 9028 in https://github.com/scality/Zenko/pull/1401)

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+        "includeAll": true,
+        "label": "env",
+        "multi": true,
+        "name": "env",
```
is this the right name to define the pod we want metrics from ?

jbertran replied:
No, indeed, this should be `pod` I guess.

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+            "selected": false,
+            "text": "1h",
+            "value": "1h"
+          },
+          {
+            "selected": false,
+            "text": "6h",
+            "value": "6h"
+          },
+          {
+            "selected": false,
+            "text": "12h",
+            "value": "12h"
+          },
+          {
+            "selected": false,
+            "text": "1d",
+            "value": "1d"
+          },
+          {
+            "selected": false,
+            "text": "7d",
+            "value": "7d"
+          },
+          {
+            "selected": false,
+            "text": "14d",
+            "value": "14d"
+          },
+          {
+            "selected": false,
+            "text": "30d",
+            "value": "30d"
+          }
+        ],
+        "query": "1m,10m,30m,1h,6h,12h,1d,7d,14d,30d",
+        "refresh": 2,
+        "skipUrlSync": false,
+        "type": "interval"
+      }
```
Do we need this one  ? we already configured diferent time intervals and options bellow.

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+  },
+  "timezone": "browser",
+  "title": "MongoDB",
+  "uid": "AyWQt9jWk",
```
this should be removed.

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+{
+  "__inputs": [
```
This section is missing the `namespace` contant input. we want to enforce the filter on only mongodb in zenko namespace.

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+      "version": "5.0.0"
+    }
+  ],
+  "annotations": {
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": "-- Grafana --",
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
```
I think this can be removed, it will be added by grafana automatically with the appropriate values.

jbertran replied:
Fair. I'll update the readme to reflect it.

lavigne958 commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "rate(mongodb_op_counters_total{pod=~\"$env\", namespace=\"zenko\"}[$interval])",
```
I see, you should not set the namespace in the metrics but keep in the inputs, this is easier to change if we deploy zenko in a different namespace, we only change 1 occurrence.

francoisferrand replied:
we may also want to further filter by service (or similar field), so we only get mongodb pods here

francoisferrand commented on monitoring/dashboards/README.md:
```diff
@@ -0,0 +1,30 @@
+
+1. Copy the JSON model
+2. Apply the following modifications:
+   * add the `__inputs` of the original download to the model
```
would it be relevant to document what should (typically) be as `inputs` and what should be as `variables` ?
Something like :

* inputs should be used match the deployment: e.g. datasource, namespace & service name to filter on...
* variables should be used to select between nodes or instances of a services

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": "-- Grafana --",
```
should be prometheus, since we use prometheus' alert manager alerts

jbertran replied:
We're removing it entirely and letting Grafana do it.

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+      }
+    ]
+  },
+  "description": "MongoDB Prometheus Exporter Dashboard. A fork inspired from https://grafana.com/grafana/dashboards/2583 which is made to work well with https://github.com/percona/mongodb_exporter.",
```
update description to something more end-user friendly ?

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1909 @@
+  "panels": [
+    {
+      "collapsed": false,
+      "datasource": null,
```
${DS_PROMETHEUS}

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1906 @@
+        "error": null,
+        "hide": 0,
+        "includeAll": true,
+        "label": "pod",
```
for the label (and for the label only!), I used "instance", which is less specific to the way it is implemented and will most likely speak to the user...

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1906 @@
+        "type": "query",
+        "useTags": false
+      },
+      {
```
what is this second variable?
i don't think i have seen it in other dashboards, and "interval" is managed by grafana anwyay.... maybe it is not required?

lavigne958 replied:
it was a variable with a lot of different time range, they are then replaced in range vector like `rate(mongodb_metric[$interval])` this way you can adjust the precision of the metrics in live, but we don't need this.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers






The following reviewers are expecting changes from the author, or must review again:

* @francoisferrand

francoisferrand commented on monitoring/dashboards/mongodb.json:
```diff
@@ -0,0 +1,1906 @@
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": "Prometheus",
```
${DS_PROMETHEUS}

(basically, any time the `datasource` is specified...)

jbertran commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers






The following reviewers are expecting changes from the author, or must review again:

* @francoisferrand







*The following options are set:* **approve**

jbertran commented:
/wait

lavigne958 committed:
ZENKO-3737 - update dashboard

Add inputs to make namespace and job configurable by the operator if necessary.
Remove occurences of `env`
remove interval variable
update metric to check mongodb uptime
Make expression that reads uptime and number of connections an _instant_ metrics, we only want the latest value.

lavigne958 commented:
Here is the resulting dashboard:

![image](https://user-images.githubusercontent.com/15084254/143425644-29597051-53f5-4c20-bee6-dc6d6e37a55c.png)
![image](https://user-images.githubusercontent.com/15084254/143425674-543191f4-0e00-4dab-a3f0-f382f7c7689a.png)

francoisferrand commented:
@lavigne958 : One comment about the description was not adressed (maybe still on jordi's computer)... And you were bitten by the CI script that this commit introduces (thanks for testing it!), you do not have the right dashboard version 😀

lavigne958 commented:
> 

I don't understand, I don't have the right version ? 
I took the dashboard left by Jordi and updated it regarding our comments.

About the description yes I will update it to something simpler like: "MongdoDB dashboard"

lavigne958 committed:
ZENKO-3737 - update description, increase version number

Commit:
ZENKO-3737 - fix dashboard version

Signed-off-by: Alexandre Lavigne <alexandre.lavigne@scality.com>

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3737.

Goodbye jbertran.
